pydra/utils/typing.py

```diff
@@ -20,6 +21,11 @@
     # Python < 3.8
     from typing_extensions import get_origin, get_args  # type: ignore
 
+try:
+    UNION_TYPES = (ty.Union, types.UnionType)
+except AttributeError:
+    UNION_TYPES = (ty.Union,)  # Python < 3.10
```

Comment:
Favor explicit version checks over try / except, see rationales [here](https://learn.scientific-python.org/development/patterns/backports/#conditional-usage)

```suggestion
if sys.version_info >= (3, 10):
    UNION_TYPES = (ty.Union, types.UnionType)
else:
    UNION_TYPES = (ty.Union,)
```

Reply:
accepted